### PR TITLE
Support to run distribution specific glibc test

### PR DESCRIPTION
**Pull request by harish-24**
Added support to run tests from distrution glibc source

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
Depends on avocado-framework/avocado#2290.

File: toolchain/glibc.py [hunk 1]
```diff
@@ -35,10 +35,17 @@ def setUp(self):
...
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            url = 'https://github.com/bminor/glibc/archive/master.zip'
+            tarball = self.fetch_asset("glibc.zip", locations=[url], expire='7d')
+            archive.extract(tarball, self.srcdir)
+            glibc_dir = os.path.join(self.srcdir, "glibc-master")
+        elif run_type == "distro":
+            glibc_dir = os.path.join(self.srcdir, "glibc-distro")
+            if os.path.exists(glibc_dir):
+                os.makedirs(glibc_dir)
```
Comment (narasimhan-v) [inline toolchain/glibc.py]: Why creating dir when path exists ?

Comment (PraveenPenguin) [reply toolchain/glibc.py]: i think  condition is like may be typo
`if not os.path.exists(glibc_dir):
   os.makedirs(glibc_dir)`

Comment (harish-24) [reply toolchain/glibc.py]: yeah, agreed.

**Comment by narasimhan-v** [comment]
I guess this can wait to be merged until https://github.com/avocado-framework/avocado/pull/2290

**Comment by harish-24** [comment]
@PraveenPenguin @narasimhan-v Can you have a look now as avocado-framework/avocado#2338 is merged now. Thanks.

**Comment by PraveenPenguin** [comment]
@harish-24 thanks for patch @narasimhan-v thanks for ACK , merging

**Pull request merged**